Fail to Initialize ecChronos When statistics.enabled is False

Currently, by default we use metrics, but after updating spring boot and other dependencies, for some reason, when we disable metrics, the error below appears and the application fail to initialize:

```maven
Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
2025-02-07T11:32:18.532-03:00 ERROR 158138 --- [           main] o.s.b.d.LoggingFailureAnalysisReporter   : 

***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 0 of constructor in com.ericsson.bss.cassandra.ecchronos.rest.MetricsREST required a bean of type 'io.micrometer.prometheusmetrics.PrometheusMeterRegistry' that could not be found.

The following candidates were found but could not be injected:
	- User-defined bean method 'prometheusMeterRegistry' in 'MetricBeans' ignored as the bean value is null


Action:

Consider revisiting the entries above or defining a bean of type 'io.micrometer.prometheusmetrics.PrometheusMeterRegistry' in your configuration.

2025-02-07T11:32:18.534-03:00 ERROR 158138 --- [           main] c.e.b.c.e.a.spring.SpringBooter          : Failed to initialize

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.ericsson.bss.cassandra.ecchronos.rest.MetricsREST': Unsatisfied dependency expressed through constructor parameter 0: No qualifying bean of type 'io.micrometer.prometheusmetrics.PrometheusMeterRegistry' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:804) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:240) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1381) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1218) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:563) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:523) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:336) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:307) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:334) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1122) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1093) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1030) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987) ~[spring-context-6.2.2.jar:6.2.2]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627) ~[spring-context-6.2.2.jar:6.2.2]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.4.2.jar:3.4.2]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752) ~[spring-boot-3.4.2.jar:3.4.2]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439) ~[spring-boot-3.4.2.jar:3.4.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318) ~[spring-boot-3.4.2.jar:3.4.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1361) ~[spring-boot-3.4.2.jar:3.4.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1350) ~[spring-boot-3.4.2.jar:3.4.2]
	at com.ericsson.bss.cassandra.ecchronos.application.spring.SpringBooter.main(SpringBooter.java:39) ~[classes/:na]
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'io.micrometer.prometheusmetrics.PrometheusMeterRegistry' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.raiseNoMatchingBeanFound(DefaultListableBeanFactory.java:2144) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveInstance(DefaultListableBeanFactory.java:1646) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1574) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1519) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:913) ~[spring-beans-6.2.2.jar:6.2.2]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791) ~[spring-beans-6.2.2.jar:6.2.2]
	... 21 common frames omitted


Process finished with exit code 1
```

Description:

_Give us a description that can answer these questions:_

_1- Does the problem persist?_
Yes
_2- Describe how to reproduce the issue, if possible. If not, describe why._
Set statistics.enabled to false in ecc.yml

Detailed description:
• What is happening?
Application is failing to start

• What did you expect to happen?
ecChronos starting as normal

• What have you tried?
• What version of ecChronos are you using?
Master's version

• Was the problem detected during an upgrade or downgrade procedure?
Upgrade from 6.0.0 to 6.0.1

• Have you checked the ecChronos documents?
Yes

• What do YOU think is the issue (optional)?
I think you're no long allowed to create null beans, we do this when metrics are disable, as MetricsREST has a injection of the prometheus bean, it fails

• Anything else that is worth being noted (optional)?

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,6 +2,8 @@
 
 ## Version 6.0.2
 
+* Fail to Initialize ecChronos When statistics.enabled is False - Issue #869
+
 ## Version 6.0.1
 
 * Bump dependecies

--- a/rest/src/main/java/com/ericsson/bss/cassandra/ecchronos/rest/MetricsREST.java
+++ b/rest/src/main/java/com/ericsson/bss/cassandra/ecchronos/rest/MetricsREST.java
@@ -38,10 +38,9 @@ import static org.springframework.http.HttpStatus.NOT_FOUND;
 @RestController
 public class MetricsREST
 {
-    @Autowired
     private final PrometheusMeterRegistry myPrometheusMeterRegistry;
 
-    public MetricsREST(final PrometheusMeterRegistry prometheusMeterRegistry)
+    public MetricsREST(@Autowired(required = false) final PrometheusMeterRegistry prometheusMeterRegistry)
     {
         myPrometheusMeterRegistry = prometheusMeterRegistry;
     }

